test/controllers/auth_test.exs

```diff
@@ -0,0 +1,68 @@
+defmodule ResearchResource.AuthTest do
+  use ResearchResource.ConnCase
+  alias ResearchResource.Auth
+
+  setup %{conn: conn} do
+    conn =
+      conn
+      |> bypass_through(ResearchResource.Router, :browser)
+      |> get("/")
+    {:ok, %{conn: conn}}
+  end
+
+  test "Auth init function", do: assert Auth.init([repo: 1])
+
+  test "login puts the user in the session", %{conn: conn} do
+    login_conn =
+      conn
+      |> Auth.login(%ResearchResource.User{id: 123})
```

Comment:
I have found aliasing all of these to be quite nice.
At the top you could have:
```
alias ResearchResource.{Auth, Router, user}
```

And then not need to write `ResearchResource...` every time